work_queue/src/work_queue.c

```diff
@@ -4384,6 +4531,15 @@ static int receive_one_task( struct work_queue *q )
 		if( task_state_is(q, taskid, WORK_QUEUE_TASK_WAITING_RETRIEVAL) ) {
 			w = itable_lookup(q->worker_task_map, taskid);
 			fetch_output_from_worker(q, w, taskid);
+			// Shutdown worker if appropriate.
+			struct work_queue_factory_info *f;
+			if ( w->factory_name &&
```

Comment:
as above

Reply:
Fixed. And I am wondering if this could cause some problems in the heap, as we are shutting down workers as we iterate the workers table.